connectivity/filters/filters.go

```diff
@@ -22,13 +22,44 @@ import (
 	flowpb "github.com/cilium/cilium/api/v1/flow"
 )
 
+type portMap map[uint32]uint32
+
+// FlowContext can carry state from one filter to another.
+type FlowContext struct {
+	// tcpPorts is filled in when matching a wildcarded source port for a TCP SYN.
+	// Subsequent non-SYN TCP matches using the same FlowContext will match this stored port number.
+	// Keyed by the known destination port so that we can track multiple connections at the same time
+	tcpPorts portMap
+
+	// udpPorts is filled in when matching a wildcarded source port for a UDP request.
+	// Subsequent UDP matches using the same FlowContext will match this stored port number.
+	// Keyed by the known destination port so that we can track multiple connections at the same time
+	udpPorts portMap
+}
+
+// saveTCPPorts marks the given port pair as seen before
+func (fc *FlowContext) saveTCPPorts(src, dst uint32) {
+	if fc.tcpPorts == nil {
+		fc.tcpPorts = make(portMap, 1)
```

Comment:
I'd say we can probably omit the `, 1)`, not sure what the default map allocation size is.